src/jit/lsra.cpp

```diff
@@ -6373,6 +6387,24 @@ void LinearScan::unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPositio
     {
         regRec->assignedInterval = regRec->previousInterval;
         regRec->previousInterval = nullptr;
+#ifdef _TARGET_ARM_
+        // Update overlapping floating point register for TYP_DOUBLE
+        if (regRec->assignedInterval->registerType == TYP_DOUBLE)
+        {
+            assert(isFloatRegType(regRec->registerType));
+
+            regNumber  nextRegNum = REG_NEXT(regRec->regNum);
+            RegRecord* nextRegRec = getRegisterRecord(nextRegNum);
+
+            nextRegRec->assignedInterval = nextRegRec->previousInterval;
+            nextRegRec->previousInterval = nullptr;
+
+            // TODO-ARM-Throughput: For ARM, we can reuse regRec->previousInterval,
+            // if following assertion is always true.
+            // Please take a look at tryAllocateFreeReg().
+            assert(nextRegRec->assignedInterval == regRec->assignedInterval);
```

Comment:
I don't believe that this is a correct assert. One or both of the two registers may have been allocated to a `TYP_FLOAT` interval.